src/Populator/PropertyMappingPopulator.php

```diff
@@ -43,7 +43,13 @@ public function __construct(
     public function populate(object $target, object $source, ?object $ctx = null): void
     {
         try {
-            $value = $this->accessor->getValue($source, $this->sourceProperty) ?? $this->defaultValue;
+            if ('$this' !== $this->sourceProperty) {
+                $sourcePropertyValue = $this->accessor->getValue($source, $this->sourceProperty);
+            } else {
+                $sourcePropertyValue = $source;
+            }
+
+            $value = $sourcePropertyValue ?? $this->defaultValue;
```

Comment:
As `$source` cannot be `null`, I'd prefer writing it like this:

```suggestion
            $value = '$this' !== $this->sourceProperty
            ? $this->accessor->getValue($source, $this->sourceProperty) ?? $this->defaultValue
            : $source;
```